plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClient.java

```diff
@@ -221,6 +237,99 @@ public PreparedStatement getPreparedStatement(Connection connection, String sql)
         return statement;
     }
 
+    @Override
+    public List<JdbcColumnHandle> getColumns(ConnectorSession session, JdbcTableHandle tableHandle)
+    {
+        if (tableHandle.getColumns().isPresent()) {
+            return tableHandle.getColumns().get();
+        }
+        checkArgument(tableHandle.isNamedRelation(), "Cannot get columns for %s", tableHandle);
+        SchemaTableName schemaTableName = tableHandle.getRequiredNamedRelation().getSchemaTableName();
+
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            Map<String, Integer> timestampPrecisions = getTimestampPrecisions(connection, tableHandle);
+            try (ResultSet resultSet = getColumns(tableHandle, connection.getMetaData())) {
```

Comment:
would it be sufficient to override this method only?

```
protected ResultSet getColumns(JdbcTableHandle tableHandle, DatabaseMetaData metadata)
```

also, MySQL `DatabaseMetaDataUsingInfoSchema`. Maybe it does all we need?

Reply:
It's a bug in the driver - they hardcode the value to 19 (for both the info schema version and the other version). Fixed via https://github.com/mysql/mysql-connector-j/commit/696c6c4131c6eb83e1ab59b8603406293587017b (driver version 8.0.16).

Reply:
Can we perhaps upgrade the driver?

Reply:
Upgrading the driver seems to fix the DST issue